src/store/search.ts

```diff
@@ -0,0 +1,78 @@
+import { ref } from 'vue';
+import {
+  RGDResultList,
+  SearchParameters,
+  ResultsFilter,
+} from './types';
+import {
+  rgdSearch,
+} from '@/api/rest';
+
+import { Polygon, MultiPolygon } from 'geojson';
+
+export   const useMap = ref(false);
+
+export const searchLimit = ref<number>(10);
+
+export const searchOffset = ref<number>(0);
+
+export const geometryInputSelection = ref();
+
+export const specifiedShape = ref<Polygon | MultiPolygon>({ type: 'Polygon', coordinates: [] });
+
+export const drawnShape = ref<Polygon | MultiPolygon>({ type: 'Polygon', coordinates: [] });
+
+export const searchResults = ref<RGDResultList>();
+export const searchResultsTotal = ref<number>();
+export const searchParameters = ref<SearchParameters>({
+  predicate: 'intersects',
+  acquired: {
+    startDate: null,
+    endDate: null,
+  },
+});
+
+
+export const resultsFilter = ref<ResultsFilter>({
+  distance: {
+    min: null,
+    max: null,
+  },
+  instrumentation: null,
+  collections: [],
+  acquired: {
+    startDate: null,
+    endDate: null,
+  },
+  time: {
+    startTime: null,
+    endTime: null,
+  },
+});
+
+export const updateResults = async () => {
+  const collectionIDs: number[] = [];
+  // eslint-disable-next-line no-unused-expressions
+  resultsFilter.value.collections?.forEach((element) => {
+    if (element.id) {
+      collectionIDs.push(element.id);
+    }
+  });
```

Comment:
I believe this can be simplified to
```suggestion
  const collectionIDs = resultsFilter.value.collections?.filter((element) => element.id).map((element) => element.id);
```

Reply:
Unless `element.id` has a numeric type, in which case you might need a more specific test (against `null` or `undefined`, etc.).